dask_cuda/tests/test_spill.py

```diff
@@ -0,0 +1,92 @@
+from distributed.utils_test import gen_cluster
+from distributed.worker import Worker
+from distributed import wait
+from dask_cuda.device_host_file import DeviceHostFile
+import dask.array as da
+import pytest
+import os
+from zict.file import _safe_key as safe_key
+
+
+def assert_device_host_file_size(dhf, total_bytes, chunk_overhead=1024):
+    byte_sum = dhf.device_buffer.fast.total_weight + dhf.host_buffer.fast.total_weight
+    file_path = [os.path.join(dhf.disk.directory, safe_key(k)) for k in dhf.disk.keys()]
+    file_size = [os.path.getsize(f) for f in file_path]
+    byte_sum += sum(file_size)
+
+    # Allow up to chunk_overhead bytes overhead per chunk on disk
+    host_overhead = len(dhf.host) * chunk_overhead
+    disk_overhead = len(dhf.disk) * chunk_overhead
+    assert (
+        byte_sum >= total_bytes
+        and byte_sum <= total_bytes + host_overhead + disk_overhead
+    )
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        {
+            "device_memory_limit": 2e9,
+            "memory_limit": 4e9,
+            "host_target": 0.6,
+            "host_spill": 0.7,
+            "spills_to_disk": False,
+        },
+        {
+            "device_memory_limit": 1e9,
+            "memory_limit": 1e9,
+            "host_target": 0.3,
+            "host_spill": 0.3,
+            "spills_to_disk": True,
+        },
+    ],
+)
+def test_device_spill(params):
```

Comment:
It is also sometimes common to include many keywords in parametrized tests.  Here is an example:

https://github.com/dask/dask/blob/8ca553dcf701a3f5580939e9da8eba487a24edb3/dask/dataframe/tests/test_dataframe.py#L278-L279

I think that what is here is good too.  I just wanted to make sure that you were aware of the other option.

Reply:
Yes, I'm aware of that, but I wanted to be clear what each means while I was writing the code. If you prefer, I wouldn't mind changing it now.